src/canvg.js

```diff
@@ -1096,12 +1163,23 @@ function build(opts) {
 
       // font
       if (typeof ctx.font != 'undefined') {
-        ctx.font = svg.Font.CreateFont(
-          this.style('font-style').value,
-          this.style('font-variant').value,
-          this.style('font-weight').value,
-          this.style('font-size').hasValue() ? this.style('font-size').toPixels() + 'px' : '',
-          this.style('font-family').value).toString();
+        if (this.style('font').hasValue()) {
+          var font = svg.Font.ParseShorthandFont(this.style('font').value)
+          ctx.font = svg.Font.CreateFont(
```

Comment:
i'm wondering if you even need to parse it and then recombine it to a string, have you tried the following?

```js
if (this.style('font').hasValue()) { ctx.font = this.style('font').value; }
else { <original code> }
```